### ISSUE TITLE
Allow choice of language for Slack integration

### ISSUE THREAD

**Issue by despens** [loomio/loomio#8561]
Loomio is localized to multiple languages, yet the Slack integration always seems to post messages using English labels. Maybe there could be a way to set the desired chatbot language?

**Comment by robguthrie** [comment]
Oh yes, absolutely. We can add a locale select to the bot config.

**Comment by robguthrie** [comment]
I've just checked on this. The chatbot will use the language, timezone and date format of the author of the notification, or the creator of the chatbot.

Will that suit your needs?

**Comment by despens** [comment]
It won't in my particular case. I'm part of several Loomio groups that use different languages, international groups use English, local groups use the local language. If each chatbot could be configured to a locale that'd be ideal I think.

**Comment by poVoq** [comment]
Since they are linked to a specific chat channel which usually has a working language, I agree that setting the language in the webhook settings would make the most sense.

**Comment by robguthrie** [comment]
We need 3 things for the chatbot: Locale (language), timezone and date format (ie: dd-mm-yy, 24/12h). 

Currently it copies the settings of the user who is responsible for the notification (ie: created the poll or thread), or if there is no user responsible, the group creator's settings. It's a similar process for sending invitation emails to new users.

I was, just now,  going to just add support for setting language on the chatbot form, then I saw we need timezone, and date format too - and it feels a bit much?

It would be easier (for me) to use the settings of the chatbot author. (ie: a change from the above description), and if you wanted to have a different locale, then you could setup a user with that locale and create the chatbot with that user?

That's a one line change versus implementing 3 new fields in the client form, server handling stuff and chatbot model

**Comment by poVoq** [comment]
Hmm, I guess that would be a reasonable compromise to keep it simpler as the person setting up the webhook probably speaks the language of the linked chat.

**Comment by robguthrie** [comment]
Yes, and if you need you can setup a user for the purpose, then mark them as a bot (from the user profile page) so they don't affect the proposal counts

**Comment by poVoq** [comment]
So I tried this with another chat integration (xmpp) and it uses the language of the browser of the person commenting, which is quite confusing as every notification then has another language pretty much.

**Comment by robguthrie** [comment]
Yeah, that logic was crazy! What was I thinking. I've fixed it just now.

https://github.com/loomio/loomio/commit/811e27d26f7c7e342123f0ea8d2aafce07a5e282

**Comment by robguthrie** [comment]
Wont fix futher. Using chatbot author locale is good enough.
